Orange/widgets/tests/test_providers.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+
+from Orange.classification import KNNLearner
+from Orange.data import Table
+from Orange.preprocess.preprocess import Preprocess
+from Orange.widgets.utils.owlearnerwidget import OWBaseLearner
+
+
+class TestProviderMetaClass(unittest.TestCase):
+
+    def assertChannelsEqual(self, first, second, msg=None):
+        self.assertEqual(len(first), len(second), msg)
+        self.assertEqual(set(i.name for i in first),
+                         set(i[0] for i in second), msg)
```

Comment:
Not sure I understand. The second assert will fail if the two sets are not of equal length – exactly what the first assert tests.